corehq/apps/es/case_search.py

```diff
@@ -179,6 +179,8 @@ def _get_domain_from_doc(self, doc):
         `doc` can be CommcCareCase instance or dict. This util method extracts domain from doc.
         This will fail hard if domain is not present in doc.
         """
+        if not doc:
+            return None
```

Comment:
I am not actually sure if I can confidently say I am aware of all the paths. But if we delete a domain which is on sub-index then we will have a couple of manual steps to do first i.e stop writes on that index by updating the config and index deletion can be taken care of then rather than automating it?